readme: add IRC badge

josephnoir opened the pull request:
Adds a badge to join the RIOT IRC channel via a web app.

### Contribution description

The README does not have any info about the IRC channel (unless I missed it). This PR adds a badge to the README to join the channel via a webpage. Maybe this encourages newcomers to ask questions.

- Hint at the IRC channel to make it easier for newcomers to ask questions.
- Add a badge the links to the IRC channel via a https://kiwiirc.com/ which only requires entering a nickname to join.

If there was already a discussion about linking to the channel and there was a decision against it ... sorry for the noise.

Here is a link to the branch to see [what it looks like](https://github.com/josephnoir/RIOT/tree/topic/add-irc-badge).

### Issues/PRs references

None.

miri64 commented:
In the [documentation](http://doc.riot-os.org/#contribute-to-riot) is linked with an `irc://` schema (however at least my IRC client fails to connect to the provided server :-/.

josephnoir commented:
Didn't know about that link. It actually asks me what app to use. I think a web app has the advantage that newcomers have an easier time to join and don't have to setup a client first.

An alternative to the kiwiirc web client would be matrix via riot.im (https://riot.im/app/#/room/#freenode_#riot-os:matrix.org). Freenode has a web app as well (https://webchat.freenode.net/), but I don't know how to pass the channel along with the URL.

I don't have preferences, just wanted to make the channel easier to find and access.

miri64 commented:
> (however at least my IRC client fails to connect to the provided server :-/.

Ah, it links to freenode.org instead of freenode.net for some reason....

miri64 commented:
> Freenode has a web app as well (https://webchat.freenode.net/), but I don't know how to pass the channel along with the URL.

I would prefer to use the web app of freenode then, instead of "enforcing" $randomClient.

miri64 commented:
> […] , but I don't know how to pass the channel along with the URL.

Here you go: http://webchat.freenode.net/?channels=%23riot-os

bergzand commented:
> I would prefer to use the web app of freenode then, instead of "enforcing" $randomClient.

I have the same opinion on this. We discussed this a bit on IRC. I prefer a client from $SERVICE whether it is the webclient from Matrix itself and chat via Matrix or the webclient from freenode to chat with freenode. Of course they don't exclude eachother.

bergzand commented:
Small nitpick, I prefer a different colour than something that indicates build failure/passed. :)

josephnoir commented:
Changed it! (Just FYI: kiwi IRC is open source and here is there [privacy policy](https://kiwiirc.com/privacy), but wanye.)

josephnoir commented:
The RIOT logo colors are red and green as well. If I see it correctly the color can be passed as a hex value. Got a color in mind?
[blue](https://img.shields.io/badge/IRC-join%20chat%20%E2%86%92-blue.svg), [pink](https://img.shields.io/badge/IRC-join%20chat%20%E2%86%92-ff69b4.svg), [lightgrey](https://img.shields.io/badge/IRC-join%20chat%20%E2%86%92-lightgrey.svg)

bergzand commented:
Blue is nice, but I don't have a strong opinion besides from what I've mentioned before :)

miri64 commented:
Blue seems to be the color for non-build stuff, at least from what I've seen for projects. Pink might not be as  attractive to some people (though I personally wouldn't want these people in our IRC :laughing:) and gray should be reserved for "undefined result" in build results.

miri64 commented on README.md:
```diff
@@ -1,4 +1,6 @@
-[![Nightly CI status master](https://ci.riot-os.org/RIOT-OS/RIOT/master/latest/badge.svg)](https://ci.riot-os.org/RIOT-OS/RIOT/master/latest/output.html)
+[![Nightly CI status master][ci-badge]](https://ci.riot-os.org/RIOT-OS/RIOT/master/latest/output.html)
```
I would rather go for `master-ci-badge`. I think we should be prepared for a future where we might want to nightly build other branches than master. Also: Why not also move the link down with `master-ci-dump` or something?

miri64 commented on README.md:
```diff
@@ -1,4 +1,6 @@
-[![Nightly CI status master](https://ci.riot-os.org/RIOT-OS/RIOT/master/latest/badge.svg)](https://ci.riot-os.org/RIOT-OS/RIOT/master/latest/output.html)
+[![Nightly CI status master][ci-badge]](https://ci.riot-os.org/RIOT-OS/RIOT/master/latest/output.html)
+[![IRC][irc-badge]](http://webchat.freenode.net?channels=%23riot-os)
```
Why not move the link down as well?

miri64 commented:
> (Just FYI: kiwi IRC is open source and here is there privacy policy, but wayne.)

Doesn't matter IMHO. Nowadays you can expect some people not to even know about IRC, so they might assume that is the chat program we all are using. I rather forward them to the web front-end of the server we are using (because my assumption is it only goes offline if freenode goes offline, while a different client might go offline at any time before that) than somewhere else ;-).

josephnoir commented:
I struggled a bit to get the badges to render correctly, although that just turned out to be a difference between `()` and `[]`. And I ended up with this.

Changed the color to blue, prefixed the CI link with `master-`, and moved all the links to the bottom.

josephnoir commented:
But I messed something up apparently. At least the links don't seem to work.

josephnoir commented:
K. If we could only get the CI to render centered now.

miri64 commented:
> K. If we could only get the CI to render centered now.

It's [a caching problem](https://github.com/github/markup/issues/224). The problem and solutions are currently discussed on our CI mailing list ;-).

josephnoir commented:
> It's a caching problem. 

That explains why it looks different in my local markdown render.

gebart commented:
Try adding a question mark at the end of the image URL to see if that bypasses the caching

miri64 commented:
> Try adding a question mark at the end of the image URL to see if that bypasses the caching

This is solved now. The GitHub cache just needed purging. Please don't put some unnecessary hacks in the README!

josephnoir commented:
Cool :)
